server/ntb/commands/sync_topics.py

```diff
@@ -0,0 +1,402 @@
+| ----- | ---- | ------ | -------- | ------------ |
+"""
+        for topic in topics:
+            qcode = topic.get("qcode") or ""
+            name = topic.get("name") or ""
+            parent = topic.get("parent") or ""
+            wikidata = topic.get("wikidata") or ""
+            iptc_subject = topic.get("iptc_subject") or ""
+            report += f"| {qcode} | {name} | {parent} | {wikidata} | {iptc_subject} |\n"
+
+        return report
+
+    def _gen_translation_table(self, topics):
+        """Generate markdown table for IPTC Media Topics without Norwegian translations"""
+
+        report = """
+Missing Norwegian Translation:
+
+| Qcode | Name |
+| ----- | ---- |
+"""
+        for topic in topics:
+            if topic["_missing_translation"]:
+                qcode = topic.get("qcode") or ""
+                name = topic.get("name") or ""
+                report += f"| {qcode} | {name} |\n"
+
+        return report
+
+    def _gen_deviated_table(self, items):
+        """Generate markdown table items that have deviated from IPTC Media Topics"""
+
+        report = """
+Deviated Items:
+
+| Qcode | Field | Existing | IPTC |
+| ----- | ----- | -------- | ---- |
+"""
+
+        for item in items:
+            topic = item.get("topic")
+            qcode = topic.get("qcode")
+            fields = item.get("fields")
+            existing = topic.get("_existing") or {}
+            for field in fields:
+                existing_value = existing.get(field) or ""
+                iptc_value = topic.get(field) or ""
+                report += f"| {qcode} | {field} | {existing_value} | {iptc_value} |\n"
+
+        return report
+
+    def _update_vocabularies_json(self, vocabularies_json, updated_topics):
+        """Updates the vocabularies json data with the synchronised Media Topics"""
+
+        # Map the IPTC Media Topics by `qcode` for easier retrieval by qcode
+        updated_topics_by_id = {
+            topic["qcode"]: topic
+            for topic in updated_topics
+        }
+
+        for vocab in vocabularies_json:
+            if vocab.get("_id") != "topics":
+                # We only want to update the `topics` CV
+                # skip this entry
+                continue
+
+            vocab.setdefault("items", [])
+
+            # Update existing local CV items
+            for item in vocab["items"]:
+                iptc_item = updated_topics_by_id[item["qcode"]]
+
+                for field in ["name", "parent", "iptc_subject", "wikidata"]:
+                    # Only apply values to fields that are not defined in local CV
+                    # Deviations from IPTC should be submitted manually from the report
+                    item[field] = item.get(field) or iptc_item.get(field)
+
+            # Add new Media Topics to the local CV
+            # default to being enabled
+            for qcode, item in updated_topics_by_id.items():
+                if not item["_existing"]:
```

Comment:
are we sure that `_existing` is always filled? A `.get()` is not necessary here?